crates/red_knot_python_semantic/resources/mdtest/dataclasses.md

```diff
@@ -91,6 +91,125 @@ repr(C())
 C() == C()
 ```
 
+## Other dataclass parameters
+
+### `repr`
+
+A custom `__repr__` method is generated by default. It can be disabled by passing `repr=False`, but
+in that case `__repr__` is still available via `object.__repr__`:
+
+```py
+from dataclasses import dataclass
+
+@dataclass(repr=False)
+class WithoutRepr:
+    x: int
+
+reveal_type(WithoutRepr(1).__repr__)  # revealed: bound method WithoutRepr.__repr__() -> str
```

Comment:
Hmm, I wonder if our display for bound methods really should show both the class the method was fetched from, and the object to which its `self` is bound...

Although mypy and pyright both reveal less than we do; they just reveal this as simply `() -> str`

Reply:
The main case I was thinking of is simply inheritance without a method override, as in the case shown here. It's a bit confusing that even though `WithoutRepr` does not have its own `__repr__` and we are finding the `__repr__` from `object`, that we show `WithoutRepr.__repr__`, a syntax that implies a `__repr__` method defined on `WithoutRepr` class. I kind of want it to say `object.__repr__` instead, but then we would no longer be showing the type of the bound `self` object.